src/Models/Test/ModelInstantiatorTest.php

```diff
@@ -434,4 +435,57 @@ public function testDontConvertNullPropertyOnRetrieve()
             )
         );
     }
+
+    public function testDatabaseModelToDataObjectNullableNestedObjectNotNullWithRelationsGetter(): void
+    {
+        // Verify a nullable one to many relationship works when the data is defined.
+        $db_model = new ModelInstantiatorTestDBModel(123);
+        $relation = new ModelInstantiatorTestDBModel(456);
+
+        $db_model->addNullableOneRelation($relation);
+
+        $data_object = $this->model_instantiator->databaseModelToDataObject(
+            $db_model,
+            ModelInstantiatorTestObject::class,
+        );
+
+        $this->assertEquals($data_object->nullableFakeHasOneRelation->pdgaNumber, 456);
+    }
+
+    public function testDatabaseModelToDataObjectNullableNestedObjectNullWithRelationsGetter(): void
+    {
+        // Verify a nullable one to many relationship works when the data is null.
```

Comment:
Comment's off.  This verifies a nullable many-to-one relationship.  Ditto on the other two tests.